pkg/apis/core/validation/cloudprofile.go

```diff
@@ -410,3 +412,99 @@ func validateRegions(regions []core.Region, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateBastion(spec *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs     field.ErrorList
+		machineArch *string
+	)
+
+	if spec.Bastion == nil {
+		return allErrs
+	}
+
+	if spec.Bastion.MachineType != nil {
+		machineArch, allErrs = validateBastionMachineType(spec.Bastion.MachineType, spec.MachineTypes, fldPath.Child("machineType"))
```

Comment:
I think it's still safer to append the `allErrs`. Otherwise, if this code gets extended in the future (e.g., something adds to `allErrs` before this line), this can easily get lost.